Lib/ntpath.py

```diff
@@ -340,8 +340,8 @@ def isreserved(path):
 def _isreservedname(name):
     """Return true if the filename is reserved by the system."""
     # Trailing dots and spaces are reserved.
-    if name.endswith(('.', ' ')) and name not in ('.', '..'):
-        return True
+    if name[-1:] in ('.', ' '):
+        return name not in ('.', '..')
```

Comment:
I'm not sure this change is an optimisation now that #117466 landed.

Reply:
@AlexWaygood, I agree that *most* of the changes are idiomatic, but perhaps not this particular one. Anyway, it does not matter; I don't think a revert is worth the churn.

Reply:
Maybe we could create a new branch to replace as many slice comparisons with `str.startwith()` & `str.endswith()`?
Doing it whenever changes are made to a module is also possible, but could have coordination issues.

Reply:
> Maybe we could create a new branch to replace as many slice comparisons with `str.startwith()` & `str.endswith()`?

Unless it would measurably improve performance on the CPython `main` branch, I don't think it would be worth the churn. (And even if it did improve performance, I'm not sure it would really be the best use of time unless it was a really significant performance improvement.)

Honestly, this is just a minor stylistic regression on one line. I really don't think we need to stress about it :-) and if there's anyone "to blame" for it, it's me, since I merged the PR before the conversation about possible optimizations to `str.startswith()` had concluded. (Since merging the PR was not urgent, I should probably have waited.)